service/discovery/openstack/openstack.go

```diff
@@ -190,46 +210,60 @@ func NewAuthorizer() (gophercloud.AuthOptions, error) {
 }
 
 // List discovers the following OpenStack resource types and translates them into the Clouditor ontology:
-// * Domains
-// * Projects
-// * Network interfaces
 // * Servers
+// * Network interfaces
 // * Block storages
+// * Domains
+// * Projects
 func (d *openstackDiscovery) List() (list []ontology.IsResource, err error) {
-	// Discover domains resource
-	domains, err := d.discoverDomains()
-	if err != nil {
-		return nil, fmt.Errorf("could not discover domains: %w", err)
+	var (
+		servers  []ontology.IsResource
+		networks []ontology.IsResource
+		storages []ontology.IsResource
+		projects []ontology.IsResource
+		domains  []ontology.IsResource
+	)
+
+	if err = d.authorize(); err != nil {
+		return nil, fmt.Errorf("could not authorize openstack: %w", err)
 	}
-	list = append(list, domains...)
 
-	// Discover project resources
-	projects, err := d.discoverProjects()
+	// First, we need to discover the resources to obtain the domain and project ID. Domains and projects are discovered last, or they are set manually if discovery is not possible due to insufficient permissions. Currently, application credentials in OpenStack are always created for a specific project within a specific domain, making discovery essentially unnecessary. The code will be retained in case this changes in the future.
+
+	// Discover servers
+	servers, err = d.discoverServer()
 	if err != nil {
-		return nil, fmt.Errorf("could not discover projects: %w", err)
+		return nil, fmt.Errorf("could not discover servers: %w", err)
```

Comment:
Kinda hard to see what has changed actually. Seems like you were just re-ordering the different discoverX calls

Reply:
Yes, it was a reordering.